### PR TITLE
Make pretty printing of map k/v's clearer

### PR DESCRIPTION
**Pull request by axman6**
This commit changes the `indef` combinator so it accepts the pretty printer of what it is indefinitely looking for - this mean that maps can annotate when a parsed value is a key or a value:

```haskell
Prelude.putStrLn . prettyHexEnc . encode $
    ( True
    , [1,2,3::Int]
    , (MP.fromList [("Hello",True),("World",False)]
    , "This is a very long sentense which should wrap"
    )
```
```

83  	# list(3)
   f5  	# bool(true)
   9f  	# list(*)
      01  	# int(1)
      02  	# int(2)
      03  	# int(3)
   ff  	# break
   82  	# list(2)
      a2  	# map(2)
         65 48 65 6c 6c 6f  	# text("Hello") [end map key]     <===
         f5  	# bool(true) [end map value]
         65 57 6f 72 6c 64  	# text("World") [end map key]
         f4  	# bool(false) [end map value]
      78 2e 54 68 69 73 20 69 73 20 61 20 76 65 72 79 
      20 6c 6f 6e 67 20 73 65 6e 74 65 6e 73 65 20 77 
      68 69 63 68 20 73 68 6f 75 6c 64 20 77 72 61 70  	# text("This is a very long sentense which should wrap")
```

### CODE REVIEW

**Comment by axman6** [comment]
This might not be very well thought through actually, if you have nested maps, it can get quite ugly and hard to follow. I'm working on another version which seems a bit clearer. New format looks like:

```haskell
Prelude.putStrLn . prettyHexEnc . encode $
    ( True,[1,2,3::Int]
    , (MP.fromList [("Hello",MP.singleton (1::Int) "some string")
                   ,("World",MP.singleton 2 "another string")]
      ,"This is a very long sentense which should wrap new line!"
      )
    )
```
```
83  # list(3)
   f5  # bool(true)
   9f  # list(*)
      01  # int(1)
      02  # int(2)
      03  # int(3)
   ff  # break
   82  # list(2)
      a2  # map(2)
          # key
         65 48 65 6c 6c 6f  # text("Hello")
          # value
         a1  # map(1)
             # key
            01  # int(1)
             # value
            6b 73 6f 6d 65 20 73 74 72 69 6e 67  # text("some string")
          # key
         65 57 6f 72 6c 64  # text("World")
          # value
         a1  # map(1)
             # key
            02  # int(2)
             # value
            6e 61 6e 6f 74 68 65 72 20 73 74 72 69 6e 67  # text("another string")
      78 38 54 68 69 73 20 69 73 20 61 20 76 65 72 79 
      20 6c 6f 6e 67 20 73 65 6e 74 65 6e 73 65 20 77 
      68 69 63 68 20 73 68 6f 75 6c 64 20 77 72 61 70 
      20 6e 65 77 20 6c 69 6e 65 21  # text("This is a very long sentense which should wrap new line!")
```

**Comment by thoughtpolice** [comment]
Cool, thanks for this! Just let me know when you're ready for a review.

**Comment by axman6** [comment]
@thoughtpolice I think I'm pretty happy with the code now, the way the markers for the keys and values line up works very well, it's clear which map a specific key/value is a part of. The only other thing I'd love to add in the future is some extra context around tags - `# tag(0)[time string]`, `# tag(64)[uint8 Typed Array]` etc.
